plugins/pulp_rpm/plugins/profilers/yum.py

```diff
@@ -87,12 +87,24 @@ def install_units(consumer, units, options, config, conduit):
                                        the specified errata
         """
         translated_units = []
+        available_rpm_nevras = None
+
         for unit in units:
             if unit['type_id'] == TYPE_ID_RPM:
                 translated_units.append(unit)
             elif unit['type_id'] == TYPE_ID_ERRATA:
+                # Populate the available RPM cache if needed. Check
+                # specifically for "None" since [] could be valid in the case
+                # of no available RPMs
+                if available_rpm_nevras is None:
+                    available_rpm_nevras = []
+                    for bound_repo_id in conduit.get_bindings(consumer.id):
+                        for r in conduit.get_repo_units(bound_repo_id, TYPE_ID_RPM):
```

Comment:
Is this going to iterate through every rpm in every repo that the consumer is bound to? If I'm updating 5000 machines in a short outage window, and we have to load all of the RPMs each machine has available, this sounds like it could thrash the database pretty hard. Could we turn it around, and search the database only for the rpms that are listed in the errata?